src/main/java/io/imply/druid/example/calcite/aggregation/ExampleSumSqlAggregator.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.druid.sql.calcite.aggregation.SqlAggregator;
+import org.apache.druid.sql.calcite.expression.DruidExpression;
+import org.apache.druid.sql.calcite.expression.OperatorConversions;
+import org.apache.druid.sql.calcite.planner.Calcites;
+import org.apache.druid.sql.calcite.planner.PlannerContext;
+import org.apache.druid.sql.calcite.rel.InputAccessor;
+import org.apache.druid.sql.calcite.rel.VirtualColumnRegistry;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * This class serves as sql binding for EXAMPLE_SUM.
+ *
+ * It defines a {@link SqlAggFunction} for EXAMPLE_SUM which takes a Numeric operand as input
+ * and converts the call to druid aggregation using {@link ExampleSumAggregatorFactory}
+ */
+public class ExampleSumSqlAggregator implements SqlAggregator
+{
+  public static final String NAME = "EXAMPLE_SUM";
+  public static final SqlAggFunction FUNCTION_INSTANCE =
+      OperatorConversions.aggregatorBuilder(NAME)
+                         .operandTypes(SqlTypeFamily.NUMERIC)
+                         .requiredOperandCount(1)
+                         .literalOperands(1)
+                         .returnTypeInference(ReturnTypes.ARG0)
+                         .functionCategory(SqlFunctionCategory.USER_DEFINED_FUNCTION)
+                         .build();
+
+  /**
+   * @return the user defined {@link SqlAggFunction}
+   */
+  @Override
+  public SqlAggFunction calciteFunction()
+  {
+    return FUNCTION_INSTANCE;
+  }
+
+  /**
+   * converts the call to an Aggregation
+   */
+  @Nullable
+  @Override
+  public Aggregation toDruidAggregation(
+      final PlannerContext plannerContext,
+      final VirtualColumnRegistry virtualColumnRegistry,
+      final String name,
+      final AggregateCall aggregateCall,
+      final InputAccessor inputAccessor,
+      final List<Aggregation> existingAggregations,
+      final boolean finalizeAggregations
+  )
+  {
+    if (aggregateCall.isDistinct()) {
+      return null;
+    }
+
+    final List<DruidExpression> arguments = Aggregations.getArgumentsForSimpleAggregator(
+        plannerContext,
+        aggregateCall,
+        inputAccessor
+    );
+
+    if (arguments == null) {
+      return null;
+    }
+
+    // we expect only one argument to the example_sum function
+    final DruidExpression arg = Iterables.getOnlyElement(arguments);
+    final String fieldName;
+
+    if (arg.isDirectColumnAccess()) {
+      fieldName = arg.getDirectColumn();
+    } else {
+      fieldName = virtualColumnRegistry.getOrCreateVirtualColumnForExpression(arg, aggregateCall.getType());
+    }
+
+    return getAggregation(name, aggregateCall, fieldName);
+  }
+
+  @Nullable
+  Aggregation getAggregation(
+      String name,
+      AggregateCall aggregateCall,
+      String fieldName
+  )
+  {
+    final ColumnType valueType = Calcites.getColumnTypeForRelDataType(aggregateCall.getType());
+    if (valueType == null) {
+      return null;
+    }
```

Comment:
what does it do?

Reply:
return type of `getColumnTypeForRelDataType` is a `Nullable` value. So having a null check to bailout. 
Added a numeric column check after this.